src/Microsoft.Diagnostics.ExtensionCommands/ClrRuntimeCommandBase.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    public abstract class ClrRuntimeCommandBase : CommandBase
+    {
+        public const string RuntimeNotFoundMessage = "No CLR runtime found. This means that a .NET runtime module or the DAC for the runtime can not be found or downloaded.";
+
+        [ServiceImport(Optional = true)]
+        public ClrRuntime Runtime { get; set; }
+
+        public override void Invoke()
+        {
+            if (Runtime == null)
+            {
+                throw new CommandNotFoundException(RuntimeNotFoundMessage);
+            }
+            ExtensionInvoke();
+        }
+
+        public abstract void ExtensionInvoke();
```

Comment:
Since the filtering for these base classes will keep Invoke from being called when the runtime == null, The whole ExtensionInvoke override with the null check is redundant. The whole ExtensionInvoke override can be removed and each command overrides Invoke. I'll make that change.